cloudflare/resource_cloudflare_access_group_test.go

```diff
@@ -0,0 +1,359 @@
+	var before, after cloudflare.AccessGroup
+	var initialID string
+	name := "cloudflare_access_group.test"
+	resourceName := strings.Split(name, ".")[1]
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckCloudflareAccessGroupDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCloudflareAccessGroupConfigBasic(resourceName, accountID, email),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckCloudflareAccessGroupExists("cloudflare_access_group.test", &before),
+					testAccManuallyDeleteAccessGroup("cloudflare_access_group.test", &initialID),
+				),
+				ExpectNonEmptyPlan: true,
+			},
+			{
+				Config: testAccCheckCloudflareAccessGroupConfigNewValue(name, accountID, email),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckCloudflareAccessGroupExists("cloudflare_access_group.test", &after),
+					testAccCheckCloudflareAccessGroupRecreated(&before, &after),
+					resource.TestCheckResourceAttr(
+						"cloudflare_access_group.test", "account_id", accountID),
+					resource.TestCheckResourceAttr(
+						"cloudflare_access_group.test", "name", fmt.Sprintf("%s/updated", name)),
+					resource.TestCheckResourceAttr(
+						"cloudflare_access_group.test", "include.email", email),
+					resource.TestCheckResourceAttr(
+						"cloudflare_access_group.test", "exclude.email", email),
+					resource.TestCheckResourceAttr(
+						"cloudflare_access_group.test", "require.email", email),
+				),
+			},
+		},
+	})
+}
+
+func TestAccCloudflareAccessGroup_UpdatingAccountIDForcesNewResource(t *testing.T) {
+	var before, after cloudflare.AccessGroup
+	name := "cloudflare_access_group.test"
+	resourceName := strings.Split(name, ".")[1]
+	newAccountID := "01a7362d577a6c3019a474fd6f485634"
+
+	resource.Test(t, resource.TestCase{
+		PreCheck: func() {
+			testAccPreCheck(t)
+			testAccPreCheckAltDomain(t)
+		},
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCloudflareAccessGroupConfigBasic(resourceName, accountID, email),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckCloudflareAccessGroupExists("cloudflare_access_group.test", &before),
+					resource.TestCheckResourceAttr("cloudflare_access_group.test", "account_id", accountID),
+				),
+			},
+			{
+				Config: testAccCloudflareAccessGroupConfigBasic(resourceName, newAccountID, email),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckCloudflareAccessGroupExists("cloudflare_page_rule.test", &after),
+					testAccCheckCloudflareAccessGroupRecreated(&before, &after),
+					resource.TestCheckResourceAttr("cloudflare_page_rule.test", "account_id", newAccountID),
+				),
+			},
+		},
+	})
+}
+
+func testAccCloudflareAccessGroupConfigBasic(resourceName, accountID, email string) string {
+	return fmt.Sprintf(`
+				resource "cloudflare_access_group" "%[1]s" {
```

Comment:
Can you format this JSON payload to be more sane with the whitespace? I know golang doesn't lend itself by default to this neatly but a two space indent is 👌